varbinview builder buffer deduplication

onursatici opened the pull request.

onursatici committed:
varbinview builder buffer deduplication

Signed-off-by: Onur Satici <onur@spiraldb.com>

robert3005 commented:
should this be optional feature of the builder? Not sure you care all the time to have it deduplicated

onursatici committed:
make deduplication optional

Signed-off-by: Onur Satici <onur@spiraldb.com>

onursatici committed:
reword

Signed-off-by: Onur Satici <onur@spiraldb.com>

robert3005 commented:
I wonder where the slowdown comes from if we kept the default the same?

onursatici committed:
uglier but faster code

Signed-off-by: Onur Satici <onur@spiraldb.com>

onursatici commented:
@robert3005 I believe it was because of the extra branch I added per view, now moved it to once per `extend_from_array` call here: https://github.com/vortex-data/vortex/pull/4036/commits/28b120788fa1a1696c4e19d989694a10ab1c4f27
